example/ck_tile/03_gemm/gemm_mem_pipeline.cpp

```diff
@@ -46,12 +46,21 @@ float gemm_calc(const gemm_basic_args& args, const ck_tile::stream_config& s)
     using TilePartitioner = ck_tile::GemmTilePartitioner<GemmShape>;
 
     using GemmEpilogue = ck_tile::Default2DEpilogue<
-        ck_tile::Default2DEpilogueProblem<AccDataType, CDataType, false, kPadC>>;
-
-    using Traits = ck_tile::TileGemmTraits<kPadA, kPadB, kPadC, ALayout, BLayout, CLayout>;
-
-    using BaseGemmPipeline = ck_tile::BaseGemmPipelineAgBgCrMem<
-        ck_tile::GemmPipelineProblem<ADataType, BDataType, AccDataType, GemmShape, Traits>>;
+        ck_tile::Default2DEpilogueProblem<AccDataType, CDataType, kPadM, kPadN>>;
+
+    using Traits = ck_tile::TileGemmTraits<kPadM,
```

Comment:
why do you move all the template paremters into `TileGemmTraits<>`. It makes the `GemmPipelineProblem<>` looks redundant.

Reply:
Because the padding, alignment, layout and data type needs to be in the same scope. But I am thinking of the grouped gemm and batched gemm or other gemm version will have the same pipeline problem. But could have different traits to switch that. If you think it is not necessary, I could substitute the problem or the traits.